### PR TITLE
Cache instance page

### PR DESCRIPTION
**Pull request by vblagoje**
Preview for @ryanemerson . Please review :-)

### CODE REVIEW

**Comment by vblagoje** [comment]
@ryanemerson I decided *not* to use sub views as the menu action buttons are completely different on these two pages. I implemented it the way we had it before. Do not see any major downsides to this approach. See if you can find some. Thank you

File: src/services/cache/CacheService.ts [hunk 1]
```diff
@@ -79,6 +92,123 @@ export class CacheService {
     return this.dmrService.readResource(request);
   }
 
+  getCacheStats(container: ICacheContainer, cache: ICache): ng.IPromise<any> {
+    let firstServer: IServerAddress = container.serverGroup.members[0];
+    let address: string [] = [].concat("host", firstServer.host, "server", firstServer.name, "subsystem",
+      "datagrid-infinispan", "cache-container", container.name, cache.type, cache.name);
```
Comment (ryanemerson) [inline src/services/cache/CacheService.ts]: Minor, but let's make use of the generateAddress function here, like you have in `getCacheUsingType(..)` as this function will eventually return a different address when running in Standalone mode. The same applies to the other request addresses.

File: src/services/cache/CacheService.ts [hunk 2]
```diff
@@ -79,6 +92,123 @@ export class CacheService {
...
+      return this.dmrService.executePost({
+        operation: "cache-rebalance",
+        address: [].concat("host", coord.host, "server", coord.name,
+          "subsystem", "datagrid-infinispan", "cache-container", container.name, cache.type, cache.name),
+        value: rebalance.toString()
+      });
+    });
+  }
+
+  private getAllCachesInContainerShallow(container: string, profile?: string): ng.IPromise<ICache[]> {
```
Comment (ryanemerson) [inline src/services/cache/CacheService.ts]: This method doesn't appear to be used anywhere

File: src/services/cache/CacheService.ts [hunk 3]
```diff
@@ -70,6 +74,15 @@ export class CacheService {
     return deferred.promise;
   }
 
+  getCacheUsingType(name: string, type: string, container: string, profile?: string): ng.IPromise<ICache> {
```
Comment (ryanemerson) [inline src/services/cache/CacheService.ts]: It seems that `getCache(..)` is now redundant (I couldn't find any usages and I think it's broken anyway). So we should remove `getCache(..)` all together and maybe rename this to `getCache()`

**Comment by ryanemerson** [comment]
@vblagoje Overall this looks good, but there are errors thrown when loading the nodes page that need fixing. 

There is also errors thrown from the server when executing the "reset statistics" operations. 

Another minor thing. It would be nice if the navigation links at the top of the page also included the caches page. E.g. `Cache containers >> cache-container >> caches >> default (distributed-cache)

File: src/services/cache/CacheService.ts (hunk 1 above)
Comment (vblagoje) [inline src/services/cache/CacheService.ts]: +1 makes sense

Comment (vblagoje) [inline src/services/cache/CacheService.ts]: But @ryanemerson but generateAddress is using profile path and here we need server path!

File: src/services/cache/CacheService.ts (hunk 2 above)
Comment (vblagoje) [inline src/services/cache/CacheService.ts]: +1 Correct @ryanemerson

File: src/services/cache/CacheService.ts (hunk 3 above)
Comment (vblagoje) [inline src/services/cache/CacheService.ts]: Corrected @ryanemerson

File: src/services/cache/CacheService.ts (hunk 1 above)
Comment (ryanemerson) [inline src/services/cache/CacheService.ts]: O yeah, I missed that, maybe we should create a generic function to create the server path then? Also, does it matter which node we get the stats from? AFAIK, it shouldn't but wasn't sure if going via the coordinator would make any difference.

**Comment by vblagoje** [comment]
@ryanemerson I updated PR. I did not change navigations links (except enabling cache containers link) as I think we can deal with this separately later. Have another look when you get a chance?

**Comment by ryanemerson** [comment]
@vblagoje Integrated 3b073182e79456a968363944eb0306a63cf07982, thanks :)

**Pull request closed without merging**